src/ngraph/op/gather.cpp

```diff
@@ -0,0 +1,77 @@
+//*****************************************************************************
+// Copyright 2019 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#include "ngraph/op/gather.hpp"
+#include "ngraph/shape.hpp"
+
+using namespace std;
+using namespace ngraph;
+
+static int PARAMS = 0;
+static int INDICES = 1;
+
+shared_ptr<Node> op::Gather::copy_with_new_args(const NodeVector& new_args) const
+{
+    check_new_args_count(this, new_args);
+    return make_shared<Gather>(new_args.at(PARAMS), new_args.at(INDICES), m_axis);
+}
+
+void op::Gather::validate_and_infer_types()
```

Comment:
I think there will need to be some element type restriction on `INDICES`, right?